libsolidity/interface/Callgraph.h

```diff
@@ -0,0 +1,148 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <libdevcore/CommonData.h>
+
+#include <set>
+
+namespace dev
+{
+namespace solidity
+{
+
+class Declaration;
+class VariableDeclaration;
+class ASTNode;
+
+struct CallgraphNode
+{
+	/// Variable declarations this node writes to, unknown if it contains a nullptr.
+	/// Note that due to aliasing, this might be incorrect for reference types.
+	std::set<VariableDeclaration const*> writes;
+	/// Variable declaration this node reads from, unknown if it contains a nullptr.
+	/// Note that due to aliasing, this might be incorrect for reference types.
+	std::set<VariableDeclaration const*> reads;
+	/// Roughly, a node has side effects (at statement level) if it cannot be removed
+	/// from the code without changing semantics. In particular, if a node
+	/// performs any modifications to any state, it has side effects.
+	/// Assignments to local variables count as side-effects,
+	/// usage of temporary memory does not count as a side-effect (unless this memory
+	/// is used elsewhere).
+	/// If any modification to the state is captured only by the value of the
+	/// expression, it does not have side effects.
+	bool hasSideEffects = false;
+	/// Reads from storage (of any account).
+	bool readsStorage = false;
+	/// Writes to storage (of any account).
+	bool writesStorage = false;
+	bool writesLogs = false;
+	/// Reads anything from the environment inculding:
+	///  - balances
+	///  - coinbase (block.coinbase)
+	///  - timestamp (block.timestamp / now)
+	///  - difficulty (block.difficulty)
+	///  - block number (block.number)
+	///  - gas limit (block.gaslimit)
+	///  - origin (tx.origin)
+	///  - gas (tx.gas)
+	///  - gas price (tx.gasprice)
+	///
+	/// but excluding:
+	///  - sender (msg.sender)
```

Comment:
Unsure about `sender`.

Whatever is decided here please also update the documentation to match.

Reply:
You convinced me, `eth_call` should not require you to specify a sender or sign the transaction.